File: cases/taint/go/recursive-carry-negative/recursive_carry_negative.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: recursive-carry-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: recursive-carry-sink

func carry(value string, depth int) string {
	if depth == 0 {
		return "clean" // DFB-WITNESS: recursive-carry-base
	}
	return carry(value, depth-1)
}

func run() {
	dfb_sink(carry(dfb_source(), 5))
}

File: cases/taint/csharp/recursive-carry-negative/RecursiveCarryNegative.cs
namespace DataFlowBench;

static class RecursiveCarryNegative
{
    static string dfb_source() // DFB-SOURCE: recursive-carry-input
    {
        return "tainted";
    }

    static void dfb_sink(string value) { } // DFB-SINK: recursive-carry-sink

    static string Carry(string value, int depth)
    {
        if (depth == 0)
        {
            return "clean"; // DFB-WITNESS: recursive-carry-base
        }

        return Carry(value, depth - 1);
    }

    static void Run()
    {
        dfb_sink(Carry(dfb_source(), 5));
    }
}
